shpc/client/__init__.py

```diff
@@ -324,6 +324,60 @@ def get_parser():
         action="store_true",
     )
 
+    # Upgrade a software to its latest version
+    upgrade = subparsers.add_parser(
+        "upgrade",
+        description=help.upgrade_description,
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    upgrade.add_argument(
+        "upgrade_recipe", help="software to upgrade", nargs="?",)
+    
+    upgrade.add_argument(
+        "--all",
+        "-a",
+        help="upgrade all installed software",
+        dest="upgrade_all",
+        action="store_true",
+    )
+
+    upgrade.add_argument(
+        "--dry-run",
+        "-d",
+        dest="dry_run",
+        help="do a dry run to view a list of the latest versions avaialble for the user's outdated software without upgrading them.",
+        action="store_true",
+    )
+    
+    upgrade.add_argument(
+        "container_image",
+        help="path to an existing container image for this software",
+        nargs="?",
+    )
```

Comment:
Remove this argument. For `shpc install` with` container_image`, it will install the module from this container.

For `shpc upgrade`, we will only use the containers from the recipe file.